src/System.Text.RegularExpressions/tests/Regex.UnicodeChar.Tests.cs

```diff
@@ -0,0 +1,202 @@
+            for (int i = 0; i < MaxUnicodeRange; i++)
+            {
+                char c = (char)i;
+                switch (CharUnicodeInfo.GetUnicodeCategory(c))
+                {
+                    case UnicodeCategory.UppercaseLetter:        //Lu
+                    case UnicodeCategory.LowercaseLetter:        //Li
+                    case UnicodeCategory.TitlecaseLetter:        // Lt
+                    case UnicodeCategory.ModifierLetter:         // Lm
+                    case UnicodeCategory.OtherLetter:            // Lo
+                    case UnicodeCategory.DecimalDigitNumber:     // Nd
+                                                                 //                    case UnicodeCategory.LetterNumber:           // ??
+                                                                 //                    case UnicodeCategory.OtherNumber:            // ??
+                    case UnicodeCategory.NonSpacingMark:
+                    //                    case UnicodeCategory.SpacingCombiningMark:   // Mc
+                    case UnicodeCategory.ConnectorPunctuation:   // Pc
+                        validChars.Add(c);
+                        break;
+                    default:
+                        invalidChars.Add(c);
+                        break;
+                }
+            }
+
+            // \w - we will create strings from valid characters that form \w and make sure that the regex engine catches this.
+            // Build a random string with valid characters followed by invalid characters
+            Random random = new Random(-55);
+            Regex regex = new Regex(@"\w*");
+
+            int validCharLength = 10;
+            int invalidCharLength = 15;
+
+            for (int i = 0; i < 100; i++)
+            {
+                StringBuilder builder1 = new StringBuilder();
+                StringBuilder builder2 = new StringBuilder();
+                for (int j = 0; j < validCharLength; j++)
+                {
+                    char c = validChars[random.Next(validChars.Count)];
+                    builder1.Append(c);
+                    builder2.Append(c);
+                }
+                for (int j = 0; j < invalidCharLength; j++)
+                    builder1.Append(invalidChars[random.Next(invalidChars.Count)]);
+
+                string input = builder1.ToString();
+                Match match = regex.Match(input);
+                Assert.True(match.Success);
+
+                Assert.Equal(builder2.ToString(), match.Value);
+                Assert.Equal(0, match.Index);
+                Assert.Equal(validCharLength, match.Length);
+
+                match = match.NextMatch();
+                do
+                {
+                    // This is tedious. But we report empty Matches for each of the non-matching characters!!!
+                    // duh!!! because we say so on the pattern - remember what * stands for :-)
+                    Assert.Equal(string.Empty, match.Value);
```

Comment:
Can you cleanup the comment to be,

```We get empty matches for each of the non-matching characters of input to match the * wildcard in regex pattern.```` 

or something more in third person. Thanks!